src/components/views/LocalModList.vue

```diff
@@ -393,13 +338,27 @@
             this.showingDependencyList = true;
         }
 
+        /**
+         * @returns whether the mod was uninstalled. If the mod was not uninstalled, an uninstall comfirmation is shown
+         */
```

Comment:
Almost.
So what happens here is that it checks if there are any dependants. If there are any dependants, it'll show a confirmation popup. If there are no dependants, then it just uninstalls the mod as nothing else would be affected.